OrbitQt/ConnectToStadiaWidget.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_CONNECT_TO_STADIA_WIDGET_H_
+#define ORBIT_QT_CONNECT_TO_STADIA_WIDGET_H_
+
+#include <QSettings>
+#include <QStateMachine>
+#include <QWidget>
+#include <memory>
+
+#include "ConnectionArtifacts.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitGgp/Client.h"
+#include "OrbitGgp/InstanceItemModel.h"
+#include "ui_ConnectToStadiaWidget.h"
+
+namespace orbit_qt {
+
+class ConnectToStadiaWidget : public QWidget {
+  Q_OBJECT
+
+ public:
+  explicit ConnectToStadiaWidget(QWidget* parent = nullptr);
+  // This needs to be called before this class can be used.
+  void SetConnectionArtifacts(ConnectionArtifacts* connection_artifacts);
```

Comment:
Could this not be added to the constructor?

Reply:
No it can't, because otherwise it cant be used in a ui file. If its supposed to be instantiated via the ui file, the constructor needs to have only one parameter (QWidget*)